components/locid/src/helpers.rs

```diff
@@ -29,47 +30,50 @@ impl<T> ShortVec<T> {
 
     pub fn push(&mut self, item: T) {
         *self = match core::mem::replace(self, Self::ZeroOne(None)) {
-            ShortVec::ZeroOne(None) => ShortVec::ZeroOne(Some(item)),
-            ShortVec::ZeroOne(Some(prev_item)) => ShortVec::Multi(vec![prev_item, item]),
-            ShortVec::Multi(mut items) => {
+            ShortSlice::ZeroOne(None) => ShortSlice::ZeroOne(Some(item)),
+            ShortSlice::ZeroOne(Some(prev_item)) => {
+                ShortSlice::Multi(vec![prev_item, item].into_boxed_slice())
+            }
+            ShortSlice::Multi(items) => {
+                let mut items = items.into_vec();
```

Comment:
Suggestion: I wonder if there's a way to make this `push` impl more efficient by allocating a vector with extra capacity such that `items.push` never needs to reallocate.

Reply:
`into_vec()` doesn't allocate, it uses the `Box`'s allocation and sets `capacity = len`. Moving the extra element allocation doesn't improve performance.

Reply:
Hmm, when we go back from `Vec` to `Box`, do we need to realloc in order to remove the extra capacity? Because `.push` will increase the capacity of the `Vec` to probably be more than it needs to be.

Reply:
Yeah if `push` over-allocates there will be two `reallocs`. We can call `reserve_exact(1)` before to try to avoid that.

That said, repeated `push`ing is pretty inefficient anyway.